src/radix_connect/mobile/relay_service/service.rs

```diff
@@ -0,0 +1,420 @@
+use super::super::session::*;
+use super::models::*;
+use crate::prelude::*;
+
+/// The service that interacts with the Radix Connect Relay.
+/// API docs at https://github.com/radixdlt/radix-connect-relay?tab=readme-ov-file#api-v1
+pub struct Service {
+    pub http_client: HttpClient,
+    pub encryption_scheme: EncryptionScheme,
+}
+
+impl Service {
+    pub fn new(http_client: HttpClient) -> Self {
+        Self {
+            http_client,
+            encryption_scheme: EncryptionScheme::default(),
+        }
+    }
+
+    pub fn new_with_network_antenna(
+        network_antenna: Arc<dyn NetworkAntenna>,
+    ) -> Self {
+        Self::new(HttpClient::new(network_antenna))
+    }
+}
+
+const SERVICE_PATH: &str =
+    "https://radix-connect-relay-dev.rdx-works-main.extratools.works/api/v1";
+
+impl NetworkRequest {
+    fn radix_connect_relay_request(request: Request) -> Result<Self> {
+        NetworkRequest::new_post(Url::from_str(SERVICE_PATH).unwrap())
+            .with_serializing_body(request)
+    }
+}
+
+impl Service {
+    async fn get_wallet_interaction_requests(
+        &self,
+        session: Session,
```

Comment:
The session will be injected from the outside in the component using the relay service.